modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/UniqueCompositeKeyObjectValidationRuleEngineImpl.java

```diff
@@ -125,11 +127,24 @@ private Predicate _getPredicate(
 				continue;
 			}
 
+			String value = null;
+
+			if (StringUtil.equals(
+					objectField.getBusinessType(),
+					ObjectFieldConstants.BUSINESS_TYPE_PICKLIST)) {
+
+				Map<String, Object> objectFieldProperties =
+					(Map<String, Object>)entryValues.get(objectField.getName());
+
+				value = String.valueOf(objectFieldProperties.get("key"));
+			}
+			else {
+				value = String.valueOf(entryValues.get(objectField.getName()));
+			}
+
 			Predicate uniqueCompositeKeyObjectFieldPredicate =
 				ObjectEntrySearchUtil.getUniqueCompositeKeyObjectFieldPredicate(
-					(Column<?, Object>)column, objectField.getDBType(),
-					GetterUtil.getString(
-						entryValues.get(objectField.getName())));
+					(Column<?, Object>)column, objectField.getDBType(), value);
```

Comment:
@mateussgn is this the only new code in this PR? If yes, it is approved 😄

Reply:
Yes, it is! Thank you, @carolmariaabb!